### ISSUE TITLE
Deprecation warnings in import, DiGraph.reverse

### ISSUE THREAD

**Issue by wkschwartz** [networkx/networkx#3230]
```python
(venv) $ python -Wa
Python 3.7.0 (default, Jun 29 2018, 20:13:13) 
[Clang 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import networkx as nx
.../venv/lib/python3.7/site-packages/networkx/classes/graph.py:23: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Mapping
.../venv/lib/python3.7/site-packages/networkx/classes/reportviews.py:95: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Mapping, Set, Iterable
.../venv/lib/python3.7/site-packages/networkx/classes/reportviews.py:95: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Mapping, Set, Iterable
.../Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
.../Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
.../Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
>>> nx.__version__
'2.2'
>>> g = nx.DiGraph()
>>> nx.reverse(g)
.../venv/lib/python3.7/site-packages/networkx/classes/digraph.py:1180: DeprecationWarning: G.fresh_copy is deprecated. Use G.__class__ instead
  H = self.fresh_copy()
<networkx.classes.digraph.DiGraph object at 0x112f52c88>
```

These deprecated lines of code are still in master as of the time I'm writing this:

https://github.com/networkx/networkx/blob/636afa0777b1c11922beb4b1cc91c2f217421066/networkx/classes/graph.py#L23

https://github.com/networkx/networkx/blob/636afa0777b1c11922beb4b1cc91c2f217421066/networkx/classes/reportviews.py#L95

https://github.com/networkx/networkx/blob/636afa0777b1c11922beb4b1cc91c2f217421066/networkx/classes/digraph.py#L1195

**Comment by AntoinePrv** [comment]
Hello,
I came to open a similar issue and found this one. I'm willing to make a PR for this. Is Python2 still supported?

**Comment by dschult** [comment]
Thank you -- 
Python 2 is no longer supported. So we just need to adjust the imports to use the Python 3 API.

**Comment by dschult** [comment]
Fixed in #3278 and #3171
